lib/java/com/google/android/material/tabs/TabLayout.java

```diff
@@ -716,36 +716,40 @@ public void setSelectedTabIndicatorHeight(int height) {
   }
 
   /**
-   * Set the scroll position of the tabs. This is useful for when the tabs are being displayed as
-   * part of a scrolling container such as {@link androidx.viewpager.widget.ViewPager}.
+   * Set the scroll position of the {@link TabLayout}.
    *
    * <p>Calling this method does not update the selected tab, it is only used for drawing purposes.
+   * This can be useful for when the tabs are being displayed as part of a scrolling container
+   * such as {@link androidx.viewpager.widget.ViewPager}.
    *
-   * @param position current scroll position
+   * @param position Position of the tab to scroll.
    * @param positionOffset Value from [0, 1) indicating the offset from {@code position}.
-   * @param updateSelectedText Whether to update the text's selected state.
+   * @param updateTabSelection Whether to draw the tab at the given position + positionOffset
+   *     as selected.
    * @see #setScrollPosition(int, float, boolean, boolean)
    */
-  public void setScrollPosition(int position, float positionOffset, boolean updateSelectedText) {
-    setScrollPosition(position, positionOffset, updateSelectedText, true);
+  public void setScrollPosition(int position, float positionOffset, boolean updateTabSelection) {
```

Comment:
Can you revert this change (and the comment) and all of the changes for `updateTabSelection` back to `updateSelectedText`?

I don't think `updateTabSelection` is a good name,  since the `updateSelectedTab` parameter only does determine whether or not to update the selected state of the textview of the tab and text appearance. `updateTabSelection` implies that the tab is being selected, which `setScrollPosition` does not do

Reply:
This parameter determines not just whether to update the text view, but whether to update _the whole tab_ (two tabs, to be more precise), including text view, icon, background, and custom view (if set). Perhaps `updateTabSelection` is not the best name, but it's definitely better and more correct than `updateSelectedText`, imo.

Reply:
Hmm I do agree that it is a better name, but still misleading since a selected tab means something already in TabLayout.

Maybe `updateTabAppearance` or something to indicate? It's also confusing since it doesn't include the indicator haha but maybe that's just something we clarify in the comments

Reply:
`updateTabAppearance` is an acceptable name, but it's too vague I think. How about `updateSelectedTabView` + some comments? (see my last change)